From 237579203a15caee49379311dc45e0abfed97de5 Mon Sep 17 00:00:00 2001
From: Mihai Moldovan <ionic@ionic.de>
Date: Tue, 29 Sep 2020 18:30:16 +0200
Subject: [PATCH] Enable implicitobsoleteusescolors flag.

This flag is normally enabled by default in libsolv if compiled on Fedora or
Mageia systems, but disabled for all others. Since we explicitly want to
support Fedora (and CentOS/RHEL) repositories with DNF, enable this feature for
correct resolution.
---
 libdnf/dnf-sack.cpp         | 1 +
 tests/hawkey/test_iutil.cpp | 9 +++++++++
 2 files changed, 10 insertions(+)

diff --git a/libdnf/dnf-sack.cpp b/libdnf/dnf-sack.cpp
index 38dcb9ec..c994b2b2 100644
--- a/libdnf/dnf-sack.cpp
+++ b/libdnf/dnf-sack.cpp
@@ -187,6 +187,7 @@ dnf_sack_init(DnfSack *sack)
     priv->pool = pool_create();
     pool_set_flag(priv->pool, POOL_FLAG_WHATPROVIDESWITHDISABLED, 1);
     pool_setdisttype(priv->pool, DISTTYPE_RPM);
+    pool_set_flag(priv->pool, POOL_FLAG_IMPLICITOBSOLETEUSESCOLORS, 1);
     priv->running_kernel_id = -1;
     priv->running_kernel_fn = running_kernel;
     priv->considered_uptodate = TRUE;
diff --git a/tests/hawkey/test_iutil.cpp b/tests/hawkey/test_iutil.cpp
index 0ca12800..8ce75bfb 100644
--- a/tests/hawkey/test_iutil.cpp
+++ b/tests/hawkey/test_iutil.cpp
@@ -157,6 +157,15 @@ START_TEST(test_version_split)
      */
     fail_if(-1 == pool_setdisttype(pool, DISTTYPE_RPM));
 
+    /*
+     * On "foreign" systems, the implicitobsoleteusescolors
+     * flag is turned off by default.
+     * This leads to unexpected results when working with
+     * Fedora, Mageia or CentOS repositories, so enable it
+     * forcefully.
+     */
+    pool_set_flag(pool, POOL_FLAG_IMPLICITOBSOLETEUSESCOLORS, 1);
+
     char evr[] = "1:5.9.3-8";
     char *epoch, *version, *release;
 
-- 
2.26.2

